Jenkinsfile

```diff
@@ -1,190 +1,139 @@
           }
+          echo 'Getting release tag'
+          release_tag = sh(returnStdout: true, script: "git describe --abbrev=0 --match \"v*\"").trim()
+          branchSpecifier = "refs/tags/${release_tag}"
+          echo branchSpecifier
 
-          if(cutNeeded == true) {
-            echo 'Getting release tag'
-            release_tag = sh(returnStdout: true, script: "git describe --abbrev=0 --match \"v*\"").trim()
-            branchSpecifier = "refs/tags/${release_tag}"
-            echo branchSpecifier
-
-            // checkout the tag so we're performing subsequent actions on it
-            sh "git checkout ${branchSpecifier}"
-
-            // set the service_sha to the current tag because it might not be the same as env.GIT_COMMIT
-            service_sha = sh(returnStdout: true, script: "git rev-parse HEAD").trim()
-
-            // set later stages to run since we're cutting
-            stage_build = true
-            stage_dockerize = true
-            stage_deploy = true
-          }
+          // checkout the tag so we're performing subsequent actions on it
+          sh "git checkout ${branchSpecifier}"
         }
       }
     }
     stage('Build') {
-      when { expression { stage_build } }
       steps {
         script {
           // perform any needed modifiers on the build parameters here
           sbtbuild.setRunITTest(true)
           sbtbuild.setNoSubproject(true)
-          sbtbuild.setScalaVersion(scala_version)
+          sbtbuild.setScalaVersion(env.SCALA_VERSION)
 
           // build
           echo "Building sbt project..."
           sbtbuild.build()
 
-          // If we're cutting, also publish libraries if they haven't already been published for this version
-          if(stage_cut == true) {
-            result = sh(returnStdout: true, script: "#!/bin/sh -e\ncurl -s -o /dev/null -w \"%{http_code}\" -u ${ARTIFACTORY_USER}:${ARTIFACTORY_PASSWORD} https://repo.socrata.com/artifactory/libs-release/com/socrata/coordinatorlib_${scala_version}/${sbtbuild.getServiceVersion()}/coordinatorlib_${scala_version}-${sbtbuild.getServiceVersion()}.pom").trim()
-            echo "Check for published artifact result: [${result}]"
-            if(result == "404") {
-              echo sh(returnStdout: true, script: "sbt +publish")
-            }
-          }
+          env.SERVICE_VERSION = sbtbuild.getServiceVersion()
+          // set the SERVICE_SHA to the current head because it might not be the same as env.GIT_COMMIT
+          env.SERVICE_SHA = sh(returnStdout: true, script: "git rev-parse HEAD").trim()
+          // set build description to be the same as the docker deploy tag
+          currentBuild.description = "${env.SERVICE}:${env.SERVICE_VERSION}_${env.BUILD_NUMBER}_${env.SERVICE_SHA.take(8)}"
+        }
+      }
+    }
+    stage('Publish') {
+      when {
+        expression { publishStage }
+      }
+      steps {
+        script {
+          echo sh(returnStdout: true, script: "sbt +publish")
```

Comment:
Presumably this could be done with sbtbuild, but I am not sure if it needs any setup, so I've kept the same command as before.